components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -2177,6 +2185,42 @@ private String handlePreConsent(OAuth2Parameters oauth2Params, AuthenticatedUser
         return additionalQueryParam;
     }
 
+    /**
+     * Filter out the requested claims with the user attributes.
+     *
+     * @param requestedClaims           List of requested claims metadata.
+     * @param userAttributes            Authenticated users' attributes.
+     * @param spTenantDomain            Tenant domain.
+     * @return                          Filtered claims with user attributes.
+     * @throws ClaimMetadataException   If an error occurred while getting claim mappings.
+     */
+    private List<ClaimMetaData> removeClaimsNotInUserAttributes(List<ClaimMetaData> requestedClaims,
+                                                                Map<ClaimMapping, String> userAttributes,
+                                                                String spTenantDomain) throws ClaimMetadataException {
+
```

Comment:
Check null/empty for `requestedClaims` and `userAttributes`. If either of them is null/empty return empty list of `ClaimMetaData`

Reply:
Fixed